ONSdigital/ras-rm-performance-tests
# RAS-937 Enable CSRF in performance env

# What and why?
This PR adds in the ability to extract the csrf token from the html and use it in post requests, this means csrf can be turned on in frontstage performance.
# How to test?
The easiest way to test is look at spinnaker (and run it in performance if you like) The results for the latest one can be found here (note each mornings are currently configured to run this)

![image](https://github.com/ONSdigital/ras-rm-performance-tests/assets/14179817/a834d2a4-96d6-4289-8842-8a1cb42efca4)

With the results in this [bucket](https://console.cloud.google.com/storage/browser/ras-rm-performance-20220908-locust/23-11-29-10-26;tab=objects?hl=en&project=ras-rm-performance-20220908&prefix=&forceOnObjectsSortingFiltering=false)

If you want to run it yourself, go to [concourse](https://github.com/ONSdigital/ras-rm-concourse/blob/main/infrastructure/performance.yml#L37) and adjust the cron job create and fly the pipeline. This will build out the env, once that is done run the performance environment setup pipeline in cluster-setup, it will do everything for you putting the results in the bucket above (obviously different date/time) The key with this is to also check frontstage in performance environment to make sure csrf is on. You can do that by looking at the yaml in the deployment of frontstage on GCP 
![image](https://github.com/ONSdigital/ras-rm-performance-tests/assets/14179817/0a7fcb56-80d7-4201-a5c5-e865666eaff5)

or forward porting to the actual environment and making sure it's in the html

n.b There is additional pipelines in place to use the latest config changes (Locust-latest and deploy performance config), these can be folded in when merged, I just needed to use different images and as they are built from branches I didn't want to change originally pipelines before approval.

 
# Trello
